app/screens/edit_profile/edit_profile.tsx

```diff
@@ -179,16 +197,55 @@ const EditProfile = ({
                 await setDefaultProfileImage(serverUrl, currentUser.id);
             }
 
-            if (hasUpdateUserInfo.current) {
+            // Only update user info if there are actually changes to unlocked fields
+            if (Object.keys(newUserInfo).length > 0) {
                 const {error: reqError} = await updateMe(serverUrl, newUserInfo);
                 if (reqError) {
                     resetScreenForProfileError(reqError);
                     return;
                 }
+            }
+
+            // Update custom attributes if changed and not SAML-linked
+            if (userInfo.customAttributes && enableCustomAttributes) {
+                // Create a map of custom fields for quick lookup
+                const customFieldsMap = new Map<string, CustomProfileFieldModel>();
+                customFields?.forEach((field) => {
+                    customFieldsMap.set(field.id, field);
+                });
+
+                // Only send custom attributes that have actually changed and are not SAML-linked
+                const changedCustomAttributes: CustomAttributeSet = {};
+                let hasCustomAttributeChanges = false;
+
+                if (customAttributesSet) {
+                    Object.keys(userInfo.customAttributes).forEach((key) => {
+                        const currentValue = customAttributesSet[key]?.value || '';
+                        const newValue = userInfo.customAttributes[key]?.value || '';
+                        const customAttribute = userInfo.customAttributes[key];
+                        const customField = customFieldsMap.get(customAttribute?.id);
+
+                        // Only include if value changed and field is not SAML-linked
+                        if (currentValue !== newValue && !isCustomFieldSamlLinked(customField)) {
+                            changedCustomAttributes[key] = userInfo.customAttributes[key];
+                            hasCustomAttributeChanges = true;
+                        }
```

Comment:
Duplicated code except for the currentValue !== newValue, which can be extracted into `hasChanges`.  For the else, hasChanges = true; and hasChanges for the if portion is

```
  const hasChanges = (customAttributesSet[key]?.value || '') !== userInfo.customAttributes[key]?.value || '');
```